src/devices/Ft4222/Ft4222Gpio.cs

```diff
@@ -0,0 +1,184 @@
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// GPIO driver for the FT4222
+    /// </summary>
+    public class Ft4222Gpio : GpioDriver
+    {
+        private const int PinCountConst = 4;
+
+        private SafeFtHandle _ftHandle = new SafeFtHandle();
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
+        private GpioPinMode[] _gpioDirections = new GpioPinMode[PinCountConst];
+
+        /// <inheritdoc/>
+        protected override int PinCount => PinCountConst;
+
+        /// <summary>
+        /// Store the FTDI Device Information
+        /// </summary>
+        public DeviceInformation DeviceInformation { get; internal set; }
+
+        /// <summary>
+        /// Create a FT4222 GPIO driver
+        /// </summary>
+        /// <param name="deviceNumber"></param>
+        public Ft4222Gpio(int deviceNumber = 0)
+        {
+            // Check device
+            var devInfos = FtCommon.GetDevices();
+            if (devInfos.Count == 0)
+                throw new IOException("No FTDI device available");
+
+            // Select the deviceNumber, only the last one in Mode 0 and Mode 1 can be open.
+            string strMode = devInfos[0].Type == FtDevice.Ft4222HMode1or2With4Interfaces ? "D" : "B";
+            
+            var devInfo = devInfos.Where(m => m.SerialNumber == strMode).ToArray();
+            if ((devInfo.Length == 0) || (devInfo.Length < deviceNumber))
+                throw new IOException($"Can't find a device to open GPIO on index {deviceNumber}");
+            
+            DeviceInformation = devInfo[deviceNumber];
+            // Open device
+            var ftStatus = FtFunction.FT_OpenEx(DeviceInformation.LocId, FtOpenType.OpenByLocation, out _ftHandle);
+
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed to open device {DeviceInformation.Description}, status: {ftStatus}");
+
+            ftStatus = FtFunction.FT4222_SetSuspendOut(_ftHandle, false);
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Can't initialize GPIO for device {DeviceInformation.Description} changing Suspend out mode, status: {ftStatus}");
+
+            ftStatus = FtFunction.FT4222_SetWakeUpInterrupt(_ftHandle, false);
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Can't initialize GPIO for device {DeviceInformation.Description} removing wake up interrupt, status: {ftStatus}");
+
+            for (int i = 0; i < PinCountConst; i++)
+                _gpioDirections[i] = GpioPinMode.Output;
+
+            ftStatus = FtFunction.FT4222_GPIO_Init(_ftHandle, _gpioDirections);
+
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Can't initialize GPIO for device {DeviceInformation.Description}, status: {ftStatus}");
+        }
+
+        /// <inheritdoc/>
+        protected override void ClosePin(int pinNumber)
+        {
+            if ((pinNumber < 0) || (pinNumber >= PinCountConst))
```

Comment:
I don't believe you need to check ranges here, @joperezr can you confirm?

Reply:
Just checked the code of the GpioController, correct, I don't need to check the pin validity, it's done in the GpioController. I'll remove all the tests.

Reply:
Right, so GpioController will do some validation upfront for you, that said if there is any validation that is very specific to the driver, then you would have to do that yourself.